db/db.js

```diff
@@ -379,6 +379,44 @@ async function getLocationData(locationid) {
 
 }
 
+
+//Retrieves the partcle door coreID corresponding to a locationid
+//This is required for ServerTest.js
+
+async function getDoorParticleCoreIDFromLocationID(locationid){
+    try{
+        const results = await pool.query("SELECT (door_particlecoreid) FROM locations WHERE locationid = $1", [locationid])
+        if(results == undefined){
+            console.log('Error: No row with the locationid exists')
+            return null
+        }
+        else{
+            return results.rows[0]
+        }
+    }
+    catch(e){
+        console.log(`Error running the getDoorParticleCoreIDFromLocationID query: ${e}`)
+    }
+}
+
+// Retrieves the locationid corresponding to a particle device coreID
+
+async function getLocationIDFromParticleCoreID(coreID){
+    try{
+        const results = await pool.query("SELECT (locationid) FROM locations WHERE door_particlecoreid = $1 OR radar_particlecoreid = $1", [coreID])
+        if(results == undefined){
+            console.log('Error: No location with associated coreID exists')
+            return null
+        }
+        else{
+            return results.rows[0]
```

Comment:
I still find this return value inconsistent with the name of the function. I would expect either:

A) The function is called `getLocationFromParticleCoreID`, the query is `SELECT * FROM locations...`, and the return value is the whole first row `results.rows[0]`.

OR

B) The function is called `getLocationidFromParticleCoreID`, the query is `SELECT locationid FROM locations...`, and the return value is just the valuve `results.rows[0].locationid`